package/MDAnalysis/analysis/contacts.py

```diff
@@ -773,3 +536,325 @@ def plot_qavg(self, filename=None, **kwargs):
+    * :math:`r^0_{ij}` is the distance from heavy atom i to j in the native state conformation,
+    * :math:`S` is the set of all pairs of heavy atoms $(i,j)$ belonging to residues $\theta_i$ and $\theta_j$ such that $|\theta_i - \theta_j| > 3$ and $r^0_{i,} < 4.5 \unicode{x212B}$,
+    * :math:`\beta=5 \unicode{x212B}^{-1},
+    * :math:`\lambda=1.8` for all-atom simulations
+
+
+    """
+    def __init__(self, u, selection, refgroup, method="cutoff", radius=4.5, outfile=None,
+                 start=None, stop=None, step=None, **kwargs):
+        """Calculate the persistence length for polymer chains
+
+        Parameters
+        ----------
+        u: Universe
+            trajectory
+        selection: tuple(string, string)
+            two contacting groups that change over time
+        refgroup: tuple(AtomGroup, AtomGroup)
+            two contacting groups in their reference conformation
+        radius: float, optional (4.5 Angstroms)
+            radius within which contacts exist
+        method: string
+            either 'cutoff' or 'best-hummer'
+
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+
+        Parameters for 'best-hummer' method
+        -----------------------------------
+        lambda_constant: float, optional (1.8 unitless)
+            contact is considered formed between (lambda*r0,r0)
+        beta: float, optional (5 Angstroms^-1)
+            softness of the switching function, the lower the softer
+
+        Attributes
+        ----------
+        results: list 
+            Fraction of native contacts for each frame
+        """
+
+        # check method
+        if not method in ("cutoff", "best-hummer"): 
+            raise ValueError("method has to be 'cutoff' or 'best-hummer'")
+        self._method = method
+
+        # method-specific parameters
+        if method == "best-hummer":
+            self.beta = kwargs.get('beta', 5.0)
+            self.lambda_constant  = kwargs.get('lambda_constant', 1.8)
+
+        # steup boilerplate
+        self.u = u
+        self._setup_frames(self.u.trajectory,
+                           start=start,
+                           stop=stop,
+                           step=step)
+
+        self.selection = selection
+        grA, grB = u.select_atoms(selection[0]), u.select_atoms(selection[1])
+        self.grA, self.grB = grA, grB        
+        refA, refB = refgroup
+
+        # contacts formed in reference
+        r0 = distance_array(refA.positions, refB.positions)
+        self.r0 = r0
+        self.mask = r0 < radius
+
+        self.contact_matrix = []
+        self.timeseries = []
+        self.outfile = outfile
+
+    @staticmethod
+    def load(self, filename):
```

Comment:
He can't do that. It is used in the method. The `staticmethod` decorator should be removed because this function is definitely not static, it is changing the internal state.